### PR TITLE
docs: add basic usage guide to sphinx docs

### PR DESCRIPTION
**Pull request by jameshcorbett**
I should probably add a one-sentence section about installation: ``pip install psi-j-python``. Since that's all there is to it, right?

### CODE REVIEW

**Comment by andre-merzky** [comment]
> I should probably add a one-sentence section about installation: `pip install psi-j-python`. Since that's all there is to it, right?

Not quite - the module is not yet pushed to pypi.  I would actually suggest to begin doing though, even before we get to 1.0 - it simplifies testing and integration, a lot.

File: docs/getting_started.rst [hunk 1]
```diff
@@ -0,0 +1,208 @@
...
+
+
+Who PSI/J is for
+----------------
+
+PSI/J is a library intended to make your job-launching code portable, or at
+least easier to port, across
+HPC centers. If you want your project to be able to request resources
+from one or more of Slurm, LSF, Flux, Cobalt, PBS, and your local machine,
+we think you will find that PSI/J simplifies your life considerably.
```
Comment (hategan) [inline docs/getting_started.rst:12]: Maybe "life" is a bit too broad :)

Comment (jameshcorbett) [reply docs/getting_started.rst:12]: How dare you understate the importance of PSI/J?

File: docs/getting_started.rst [hunk 2]
```diff
@@ -0,0 +1,208 @@
...
+
+
+Who PSI/J is (probably) not for
+-------------------------------
+
+If you were sure that you will only *ever* be launching jobs on ORNL's Summit
+system, and you don't care about any other cluster or machine, then you may as well
+interact with LSF (the resource manager on Summit) directly, rather than
+indirectly through PSI/J. In that case PSI/J would not really be adding much
+other than additional complexity.
```
[the remark this answers is not in the document]
Comment (jameshcorbett) [reply docs/getting_started.rst]: Good point, thanks.

**Pull request merged**